File: Part 2 - Regression/Section 9 - Random Forest Regression/random_forest_regression.R
#Random Forest regression
#Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]


#Fitting Random Forest Regression Model to dataset
library(randomForest)
#With brackets, give data frames, $ gives a vector (according to arguments)
set.seed(1234)
regressor = randomForest(x = dataset[1],
                         y = dataset$Salary,
                         ntree = 500)

#Predicting a result with Random Forest Regression
y_pred = predict(regressor, data.frame(Level = 6.5))

#Visualizing Random Forest reg (high resolution)
library(ggplot2)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)
ggplot() + 
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             colour = 'red') +
  geom_line(aes(x = x_grid, y =  predict(regressor, newdata = data.frame(Level = x_grid))),
            colour = 'blue') + 
  ggtitle('Truth or Bluff(Random Forest Regression)') +
  xlab('Level') + 
  ylab('Salary')

File: Part 2 - Regression/Section 9 - Random Forest Regression/random_forest_regression.py
#Random Forest Regression
    
#Importing libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Importing dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:,1:2].values
y = dataset.iloc[:,2].values


#Splitting data into training and test set
"""from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)"""

#Feature scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)"""


#Fitting Random Forest regression model to the dataset
from sklearn.ensemble import RandomForestRegressor
regressor = RandomForestRegressor(n_estimators=300, random_state=0)
regressor.fit(X,y)
#Predicting a new result with Poly Regression
y_pred = regressor.predict(6.5)


#Visualizing Regression results (for higher resolution and smoother curve)
X_grid = np.arange(min(X), max(X), 0.01)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, y, color = 'red')
plt.plot(X_grid, regressor.predict(X_grid), color = 'blue')
plt.title('Truth or Bluff (Random Forest Regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()